src/java/org/apache/cassandra/tools/nodetool/DataPaths.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.tools.nodetool;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.airlift.airline.Arguments;
+import io.airlift.airline.Command;
+import io.airlift.airline.Option;
+import org.apache.cassandra.tools.NodeProbe;
+import org.apache.cassandra.tools.NodeTool.NodeToolCmd;
+import org.apache.cassandra.tools.nodetool.stats.DataPathsHolder;
+import org.apache.cassandra.tools.nodetool.stats.DataPathsPrinter;
+import org.apache.cassandra.tools.nodetool.stats.StatsHolder;
+import org.apache.cassandra.tools.nodetool.stats.StatsPrinter;
+
+@Command(name = "datapaths", description = "Print all directories where data of tables are stored")
+public class DataPaths extends NodeToolCmd
+{
+    @Arguments(usage = "[<keyspace.table>...]", description = "List of table (or keyspace) names")
+    private List<String> tableNames = new ArrayList<>();
+
+    @Option(title = "format", name = {"-F", "--format"}, description = "Output format (json, yaml)")
+    private String outputFormat = "";
+
+    @Override
+    protected void execute(NodeProbe probe)
+    {
+        if (!outputFormat.isEmpty() && !"json".equals(outputFormat) && !"yaml".equals(outputFormat))
+        {
+            throw new IllegalArgumentException("arguments for -F are yaml and json only.");
+        }
+
+        StatsHolder holder = new DataPathsHolder(probe, tableNames);
+        StatsPrinter printer = DataPathsPrinter.from(outputFormat);
```

Comment:
I would renamed this method from `from` to `for`. We want to get a printer _for_ an output format, not _from_ it.

Reply:
I could refactor that, but the other implementations I used as blueprint are also using this naming:

- https://github.com/apache/cassandra/blob/d80bb323c9dd8f814fe02e16dec510557c7d5101/src/java/org/apache/cassandra/tools/nodetool/stats/TableStatsPrinter.java#L28
- https://github.com/apache/cassandra/blob/d80bb323c9dd8f814fe02e16dec510557c7d5101/src/java/org/apache/cassandra/tools/nodetool/stats/TpStatsPrinter.java#L36
- https://github.com/apache/cassandra/blob/d80bb323c9dd8f814fe02e16dec510557c7d5101/src/java/org/apache/cassandra/tools/nodetool/stats/CompactionHistoryPrinter.java#L32